addons/etcd/config/etcd.conf.yaml.tpl

```diff
@@ -0,0 +1,203 @@
+# https://github.com/etcd-io/etcd/blob/main/etcd.conf.yml.sample
+# using this config file will ignore ALL command-line flag and environment variables.
+
+{{- $peerProtocol := "http" -}}
+{{- if $.component.tlsConfig -}}
+  {{- if eq .PEER_TLS "true" -}}
+    {{- $peerProtocol = "https" -}}
+  {{- end -}}
+{{- end -}}
+
+{{- $clientProtocol := "http" -}}
+{{- if $.component.tlsConfig -}}
+  {{- if eq .CLIENT_TLS "true" -}}
+    {{- $clientProtocol = "https" -}}
+  {{- end -}}
+{{- end -}}
+
+# Human-readable name for this member.
+name: 'default'
+
+# Path to the data directory.
+data-dir: {{ .DATA_DIR }}
+
+# Path to the dedicated wal directory.
+wal-dir:
+
+# Number of committed transactions to trigger a snapshot to disk.
+snapshot-count: 10000
+
+# Time (in milliseconds) of a heartbeat interval.
+heartbeat-interval: 100
+
+# Time (in milliseconds) for an election to timeout.
+election-timeout: 1000
+
+# Raise alarms when backend size exceeds the given quota. 0 means use the
+# default quota.
+quota-backend-bytes: 0
+
+# List of comma separated URLs to listen on for peer traffic.
+listen-peer-urls: {{$peerProtocol}}://0.0.0.0:2380
+
+# List of comma separated URLs to listen on for client traffic.
+listen-client-urls: {{$clientProtocol}}://0.0.0.0:2379
+
+# Maximum number of snapshot files to retain (0 is unlimited).
+max-snapshots: 5
+
+# Maximum number of wal files to retain (0 is unlimited).
+max-wals: 5
+
+# Comma-separated white list of origins for CORS (cross-origin resource sharing).
+cors:
+
+# List of this member's peer URLs to advertise to the rest of the cluster.
+# The URLs needed to be a comma-separated list.
+initial-advertise-peer-urls: {{$peerProtocol}}://0.0.0.0:2380
+
+# List of this member's client URLs to advertise to the public.
+# The URLs needed to be a comma-separated list.
+advertise-client-urls: {{$clientProtocol}}://0.0.0.0:2379
+
+# Discovery URL used to bootstrap the cluster.
+discovery:
+
+# Valid values include 'exit', 'proxy'
+discovery-fallback: 'proxy'
+
+# HTTP proxy to use for traffic to discovery service.
+discovery-proxy:
+
+# DNS domain used to bootstrap initial cluster.
+discovery-srv:
+
+{{ define "INIT_PEERS" -}}
+  {{- $peerProtocol := "http" -}}
+  {{- if $.component.tlsConfig -}}
+    {{- if eq .PEER_TLS "true" -}}
+      {{- $peerProtocol = "https" -}}
+    {{- end -}}
+  {{- end -}}
+  {{- $POD_NAME := splitList "," .POD_NAME_LIST -}}
+  {{- $PEERS := splitList "," .PEERS -}}
```

Comment:
PEERS is pods' FQDN addresses list, it only supports inner-k8s access. 
Original etcd addon uses loadbalancer ip to support  cross-k8s access, should support it.

Reply:
done